### DISCUSSION TITLE
Proposal for the Software Programs section 4.1.2.10

### DISCUSSION THREAD

**Discussion by waddella** [RConsortium/submissions-wg#7] [General]
# Motivation

Let's start the discussion about an alternative wording proposal for section `4.1.2.10` in the "FDA STUDY DATA
TECHNICAL CONFORMANCE GUIDE" found here:

https://www.fda.gov/media/143550/download

Currently (November 2020) the paragraph reads:

> Sponsors should provide the software programs used to create all ADaM datasets and generate tables and figures associated with primary and secondary efficacy analyses. Furthermore, sponsors should submit software programs used to generate additional
information included in Section 14 CLINICAL STUDIES of the Prescribing Information, if applicable. The specific software utilized should be specified in the ADRG. Refer to FDA Statistical Software Clarifying Statement for more information.
> 
> The main purpose of requesting the submission of these programs is to understand the process by which the variables for the respective analyses were created and to confirm the analysis algorithms and results. Sponsors should submit software programs in ASCII text format. Executable file extensions should not be used.

The goal is to be more specific about the file types that can be submitted as many have interpreted that paragraph that program files need to be submitted as `.txt` files.

Am I correct that the actual constraint is that the files need to be:

* ASCII files no binaries
* `iso-8859-1` encoding and not `utf-8`
* lowercase file endings, so `.R` does not work but `.r` would not work 

Also would `.tar` be acceptable for a collection of ASCII files?

Let's start the discussion here, I can update this first message with the agreed solution from the discussion thread.

# Current discussion state

TBD

**Comment by reikoch** [comment]
I would discourage use of iso-8859-1 encodings at all; health authorities communicate with other countries' authorities, better stick to utf-8 generally for simplicity. The US-ASCII subset of utf-8 can be a compromise for submissions to FDA.
